contrib/olivere/elastic/option.go

```diff
@@ -46,3 +48,10 @@ func WithAnalyticsRate(rate float64) ClientOption {
 		cfg.analyticsRate = rate
 	}
 }
+
+// WithResourceNamer sets a function to set the span's resource name
+func WithResourceNamer(namer func(url, method string) string) ClientOption {
```

Comment:
This step here is going to lock in the "public API" of this and future `ResourceNamer`s.
It's worth considering any extra data that might be useful to a `ResourceNamer` even if `quantize` and `staticNamer` don't currently need it.
Changing the function signature in the future will be trickier.

Reply:
That's a good point. What about passing the entire request into the `ResourceNamer`?

Reply:
A copy of it might be OK.
Passing the original would be risky if it were modified or things like the request body were read incorrectly.
I'd wait for thoughts/suggestions from @gbbr before making code changes though.

Reply:
I'm not entirely sure about that. I'd say we start simple. My concerns with copying the context are performance related. I'm also not entirely convinced that there is anything to be gained by passing more information than the `url` and `method`. I'd vote for not doing that yet. If we learn later that this decision was wrong, we can set it as a proposal for [v2.0](https://github.com/DataDog/dd-trace-go/labels/v2.0).